src/main/java/seedu/address/ui/RightSideBar.java

```diff
@@ -54,11 +57,21 @@ public void update(String commandText) {
      * Filters and returns the requiredPeriod according to the commandText
      */
     private String requiredPeriod(String commandText) {
-        if (commandText.contains(PREFIX_PERIOD.toString())) {
-            int startOfPeriod = commandText.indexOf(PREFIX_PERIOD.toString());
-            return commandText.substring(startOfPeriod + 2).toUpperCase();
+        String firstWord = commandText.split(" ")[0];
+        if (firstWord.equals(ViewSessionCommand.COMMAND_WORD)) {
```

Comment:
" sview p/all" will not update the label

Reply:
will the updated change work?

Reply:
will the updated change work